Prevent downstream dependencies to pickup MinVer as a depencency

romerod opened the pull request:
## Proposed Changes

MinVer is beeing picked up by downstream dependencies which it shouldn't, see https://github.com/adamralph/minver/blob/main/README.md?plain=1#L55 .

This even breaks compatibility of the package with NonSDK projects.

https://www.nuget.org/packages/RabbitMQ.Client.OAuth2 was already published with this dependency.

As I didn't change any code I didn't run the tests.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes issue #NNNN)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask on the
mailing list. We're here to help! This is simply a reminder of what we are
going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [ ] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [ ] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in related repositories

lukebakken reviewed (commented):
Thank you @romerod

Does this apply to any of the other package references in this project as well?

paulomorgado commented:
> Thank you @romerod
> 
> Does this apply to any of the other package references in this project as well?

Any package that is development time only and not a run time dependency, should have this.

This should be automatic when adding the reference through NuGet tooling.

romerod commented:
> > Thank you @romerod
> > Does this apply to any of the other package references in this project as well?
> 
> Any package that is development time only and not a run time dependency, should have this.
> 
> This should be automatic when adding the reference through NuGet tooling.

exactly, I didn't focus on the other global references.

Did take a quick look now and at first sight they also seem to be development time dependencies.

romerod commented:
Adding PrivateAssets=True to all GlobalProjectReference's got me thinking that the https://www.nuget.org/packages/RabbitMQ.Client.OAuth2#dependencies-body-tab package didn't list the others two packages as dependencies.

So I wanted to check if they were added after the release of RabbitMQ.Client.OAuth2 2.0.0. While doing this I've discovered the following things:

- RabbitMQ.Client.OAuth2 2.0.0 was released from [RabbitMQ.Client.OAuth2-NuGet/RabbitMQ.Client.OAuth2.csproj](https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/main/projects/RabbitMQ.Client.OAuth2-NuGet/RabbitMQ.Client.OAuth2.csproj) which has <ManagePackageVersionsCentrally>false</ManagePackageVersionsCentrally>
- There is an issue https://github.com/NuGet/Home/issues/14223 saying: "GlobalPackageReference name implies that we are adding a PackageReference without any customizations to all projects. This is not the case", the [doc](https://learn.microsoft.com/en-us/nuget/consume-packages/central-package-management#global-package-references) that it adds ProjectReference with PrivateAssets=True.

So I removed the PrivateAssets=True from all GlobalProjectReference's and added it to [RabbitMQ.Client.OAuth2-NuGet/RabbitMQ.Client.OAuth2.csproj](https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/main/projects/RabbitMQ.Client.OAuth2-NuGet/RabbitMQ.Client.OAuth2.csproj)

paulomorgado commented:
`PrivateAssets=all` should be the default for `GlobalPackageReference`. Have you tried just removing the package reference from the csproj?

romerod commented:
@danielmarbach where did you see that you PrivateAssets can be an attribute?

In the doc it's a sub Element.

https://learn.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files

paulomorgado commented:
> @danielmarbach where did you see that you PrivateAssets can be an attribute?
> 
> In the doc it's a sub Element.
> 
> https://learn.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files

MSBuild item properties can be child elements or attributes.

danielmarbach commented:
What @paulomorgado said. Wonder why the verify step is suddenly failing now after that update. My changes should not change that outcome

danielmarbach commented:
It seems the .net version is not "pinned" and the build uses the version that comes with the image.

Successful build

![image](https://github.com/user-attachments/assets/1c7a05f1-c2a2-42b3-827c-1f1a616a07d0)

.NET Core SDK: 6.0.136, 6.0.203, 6.0.321, 6.0.428, 8.0.115, 8.0.206, 8.0.311, 8.0.408, 9.0.105, 9.0.200, 9.0.203

New image

![image](https://github.com/user-attachments/assets/b4ce1e4b-e620-4717-8ade-e01e83a621c6)

.NET Core SDK: 6.0.136, 6.0.203, 6.0.321, 6.0.428, 8.0.116, 8.0.206, 8.0.313, 8.0.410, 9.0.106, 9.0.200, 9.0.204, 9.0.300

which quite likely influences dotnet format rule evaluation.

danielmarbach commented:
See https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1842

michaelklishin commented:
Thank you, @romerod @danielmarbach @paulomorgado.

lukebakken commented:
Thanks everyone!

romerod commented:
When will a new version of the RabbitMQ.Client.OAuth2 package be published?
